### PR TITLE
Kong plugins feature

### PR DESCRIPTION
**Pull request by mlavacca**
**What type of PR is this?**

**What this PR does / why we need it**:

Kong plugins feature. The `konhq.com/plugins` ingress annotation is converted to a list of `ExtensionRef` filters.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
[Kong Provider] Add support for converting the `konghq.com/plugins` ingress annotation to a list of `ExtensionRef` HTTPRoute filters.
```

### CODE REVIEW

File: pkg/i2gw/providers/kong/plugins.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// pluginsFeature parses the Kong Ingress Controller plugins annotation and convert it
+// into HTTPRoutes rule's ExtensionRef filter.
+// It's possible to define a list of plugins to attach to the same HTTPRoute by setting
+// a comma separated list.
+//
+// Example: konghq.com/plugins: "plugin1,plugin2"
```
Comment (levikobi) [inline pkg/i2gw/providers/kong/plugins.go:34]:
```suggestion
// pluginsFeature parses the Kong Ingress Controller plugins annotation and converts it
// into HTTPRoutes rule's ExtensionRef filter.
// It's possible to define a list of plugins to attach to the same HTTPRoute by setting
// a comma-separated list.
//
// Example: konghq.com/plugins: "plugin1,plugin2"
```

**Review by levikobi — APPROVED** [review]
Thank you for another great contribution @mlavacca!
Just a small NIT, other than that lgtm

File: pkg/i2gw/providers/kong/plugins.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			filtersValues := strings.Split(val, ",")
+			for _, v := range filtersValues {
+				if v == "" {
+					continue
+				}
+				filters = append(filters, gatewayv1beta1.HTTPRouteFilter{
+					Type: gatewayv1beta1.HTTPRouteFilterExtensionRef,
+					ExtensionRef: &gatewayv1beta1.LocalObjectReference{
+						Group: gatewayv1beta1.Group("configuration.konghq.com/v1"),
+						Kind:  gatewayv1beta1.Kind("KongPlugin"),
```
Comment (LiorLieberman) [inline pkg/i2gw/providers/kong/plugins.go:65]: nit: consider moving these strings to constants

**Review by levikobi — COMMENTED** [review]
/lgtm

File: pkg/i2gw/providers/kong/converter_test.go [hunk 3]
```diff
@@ -131,6 +132,16 @@ func Test_ToGateway(t *testing.T) {
 										Method: ptrTo(gatewayv1beta1.HTTPMethodPost),
 									},
 								},
+								Filters: []gatewayv1beta1.HTTPRouteFilter{
+									{
+										Type: gatewayv1beta1.HTTPRouteFilterExtensionRef,
+										ExtensionRef: &gatewayv1beta1.LocalObjectReference{
+											Group: gatewayv1beta1.Group("configuration.konghq.com/v1"),
+											Kind:  gatewayv1beta1.Kind("KongPlugin"),
```
Comment (LiorLieberman) [inline pkg/i2gw/providers/kong/converter_test.go:140]: can these use the new consts as well?

Comment (mlavacca) [reply pkg/i2gw/providers/kong/converter_test.go]: yep, sorry for missing it. All the references to such an object should be using the const values now 👍

**Pull request merged**